src/ice/candidate.rs

```diff
@@ -351,6 +351,13 @@ impl Candidate {
     pub(crate) fn clear_ufrag(&mut self) {
         self.ufrag = None;
     }
+
+    pub(crate) fn replace_ufrag(&self, ufrag: &str) -> Candidate {
+        let mut new = self.clone();
+        new.set_ufrag(&ufrag);
+
+        new
+    }
```

Comment:
I think `clone()` + `set_ufrag()` would be clearer and not requiring any new function.

Reply:
Sure